scroll_derive/tests/tests.rs

```diff
@@ -158,6 +158,28 @@ fn test_nested_struct() {
     assert_eq!(b, b2);
 }
 
+#[derive(Debug, PartialEq, Eq, Pread, Pwrite, IOread, IOwrite, SizeWith)] 
+#[repr(C)]
+struct Data8<T, Y> {
+    ids: [T; 3],
+    xyz: Y
+}
+
+#[test]
+fn test_generics() {
+    let mut bytes = [0xde, 0xed, 0xef, 0x10, 0x10];
+    let data: Data8<u8, u16> = bytes.pread_with(0, LE).unwrap();
```

Comment:
what happens if we do:
```rust
let data: Data8<Data8<u8, 16>, u16> = bytes.pread_with()
```
is this just a flat packed serialize/deserialize ?

Reply:
That won't work because of the trait bounds for the generic params in the impl, but I think it would be useful if it were possible. I'll see if I can figure out a way to allow for that.

Reply:
Yea i'm just wondering what would happen; and also my refrain that this isn't meant to be a general purpose serialization/deserialization library, but i suppose that cat might have gotten out already ;)

Reply:
You'd just get an error like `the trait bound Data8<u8, 16>: std::convert::From<u8> is not satisfied` with this implementation.